src/main/kotlin/com/sourcegraph/cody/auth/SelectOneOfTheAccountsAsActive.kt

```diff
@@ -3,18 +3,24 @@ package com.sourcegraph.cody.auth
 import com.intellij.openapi.project.Project
 import com.sourcegraph.cody.CodyToolWindowContent
 import com.sourcegraph.cody.config.CodyAuthenticationManager
+import com.sourcegraph.cody.config.CodyProjectActiveAccountHolder
 import com.sourcegraph.cody.config.getFirstAccountOrNull
 import com.sourcegraph.cody.initialization.Activity
 
 class SelectOneOfTheAccountsAsActive : Activity {
 
   override fun runActivity(project: Project) {
-    if (CodyAuthenticationManager.getInstance(project).hasNoActiveAccount()) {
-      val newActiveAccount =
-          CodyAuthenticationManager.getInstance(project).getAccounts().getFirstAccountOrNull()
-      CodyAuthenticationManager.getInstance(project).setActiveAccount(newActiveAccount)
+    // It is possible that CodyProjectActiveAccountHolder loads the account from the state omitting
+    // setActiveAccount() call. We need the call to happen as it propagates the configuration
+    // changes to the agent and the account settings changes to the UI.
+    val initialAccount =
+        CodyProjectActiveAccountHolder.getInstance(project).account
+            ?: CodyAuthenticationManager.getInstance(project).getAccounts().getFirstAccountOrNull()
+    if (initialAccount == null) {
       // The call to refreshPanelsVisibility() is needed to update the UI when there is no account.
       CodyToolWindowContent.executeOnInstanceIfNotDisposed(project) { refreshPanelsVisibility() }
+    } else {
+      CodyAuthenticationManager.getInstance(project).setActiveAccount(initialAccount)
```

Comment:
Does it make ay sense to `setActiveAccount` also if active account was already set?

Reply:
Here, it can be loaded from memory but not yet actually set (not propagated to the agent, not reflected in our UI). Technically, we do often `setActiveAccount` if the active account is set when we switch the accounts.